app/components/app_session_needs_review_warning_component.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class AppSessionNeedsReviewWarningComponent < ViewComponent::Base
+  erb_template <<-ERB
+    <div class="nhsuk-warning-callout">
+      <h3 class="nhsuk-warning-callout__label">
+        <span role="text">
+          <span class="nhsuk-u-visually-hidden">Important: </span>
+          Needs review
+        </span>
+      </h3>
+
+      <ul>
+        <% if @session.patients.without_nhs_number.count > 0 %>
+          <li>
+            <%= link_to t(:children_without_nhs_number, count: @session.patients.without_nhs_number.count),
```

Comment:
```suggestion
            <%= link_to t(:children_without_nhs_number, count: patient_session.merge(Patient.without_nhs_number).count),
```

It feels like we need this value in 3 places so it might be worth memoizing it in a method.

```rb 
def without_nhs_number_count
  @without_nhs_number_count ||= ...
```

Reply:
We now cache a warnings object that contains this information